### PR TITLE
Allow custom server URLs to be set and used via ParseClient.Configuration.

### PR DESCRIPTION
**Pull request by richardjrossiii**

### CODE REVIEW

File: Parse/Public/ParseClient.cs [hunk 1]
```diff
@@ -122,7 +129,7 @@ public struct Configuration {
     /// </param>
     public static void Initialize(Configuration configuration) {
       lock (mutex) {
-        HostName = HostName ?? new Uri("https://api.parse.com/1/");
```
Comment (hallucinogen) [inline Parse/Public/ParseClient.cs]: do we still `HostName`? (even in tests). If we don't, we need to clean that up (doesn't need to be in this PR)

Comment (richardjrossiii) [inline Parse/Public/ParseClient.cs]: You're right. I can update all of our internal tests to use Configuration.Server.

**Comment by hallucinogen** [comment]
lgtmcorgi

Don't forget to update IntegrationTests

**Comment by richardjrossiii** [comment]
UnitTests now pass.

**Comment by hallucinogen** [comment]
LGTM

**Pull request merged**